arrow/src/main/java/org/apache/iceberg/arrow/vectorized/parquet/VectorizedDeltaEncodedValuesReader.java

```diff
@@ -0,0 +1,290 @@
+    if (valuesRead == 0) {
+      ByteBuffer firstValueBuffer = getBuffer(typeWidth);
+      outputWriter.write(firstValueBuffer, firstValue);
+      if (vec != null) {
+        vec.getDataBuffer().setBytes((long) rowId * typeWidth, firstValueBuffer);
+      }
+      lastValueRead = firstValue;
+      rowId++;
+      remaining--;
+    }
+    while (remaining > 0) {
+      int n;
+      try {
+        n = loadMiniBlockToOutput(remaining, vec, rowId, typeWidth, outputWriter);
+      } catch (IOException e) {
+        throw new ParquetDecodingException("Error reading mini block.", e);
+      }
+      rowId += n;
+      remaining -= n;
+    }
+    valuesRead = total - remaining;
+  }
+
+  /**
+   * Read from a mini block. Read at most 'remaining' values into output.
+   *
+   * @return the number of values read into output
+   */
+  private int loadMiniBlockToOutput(
+      int remaining, FieldVector vec, int rowId, int typeWidth, IntegerOutputWriter outputWriter)
+      throws IOException {
+
+    // new block; read the block header
+    if (remainingInBlock == 0) {
+      readBlockHeader();
+    }
+
+    // new miniblock, unpack the miniblock
+    if (remainingInMiniBlock == 0) {
+      unpackMiniBlock();
+    }
+
+    // read values from miniblock
+    ByteBuffer buffer = getBuffer(remainingInMiniBlock * typeWidth);
+    int valuesRead = 0;
+    for (int i = miniBlockSizeInValues - remainingInMiniBlock;
+        i < miniBlockSizeInValues && valuesRead < remaining;
+        i++) {
+      // calculate values from deltas unpacked for current block
+      long outValue = lastValueRead + minDeltaInCurrentBlock + unpackedValuesBuffer[i];
+      lastValueRead = outValue;
+      outputWriter.write(buffer, outValue);
+      remainingInBlock--;
+      remainingInMiniBlock--;
+      valuesRead++;
+    }
+    if (vec != null) {
+      vec.getDataBuffer().setBytes((long) rowId * typeWidth, buffer);
+    }
+
+    return valuesRead;
+  }
+
+  private void readBlockHeader() {
+    try {
+      minDeltaInCurrentBlock = BytesUtils.readZigZagVarLong(in);
+    } catch (IOException e) {
+      throw new ParquetDecodingException("Can not read min delta in current block", e);
+    }
+    readBitWidthsForMiniBlocks();
+    remainingInBlock = blockSizeInValues;
+    currentMiniBlock = 0;
+    remainingInMiniBlock = 0;
+  }
+
+  private ByteBuffer getBuffer(int length) {
+    try {
+      return this.in.slice(length).order(ByteOrder.LITTLE_ENDIAN);
+    } catch (IOException e) {
+      throw new ParquetDecodingException("Failed to read " + length + " bytes", e);
+    }
+  }
+
+  /**
+   * mini block has a size of 8*n, unpack 32 value each time
+   *
+   * <p>see org.apache.parquet.column.values.delta.DeltaBinaryPackingValuesReader#unpackMiniBlock
+   */
+  private void unpackMiniBlock() throws IOException {
+    Arrays.fill(this.unpackedValuesBuffer, 0);
+    BytePackerForLong packer =
+        Packer.LITTLE_ENDIAN.newBytePackerForLong(bitWidths[currentMiniBlock]);
```

Comment:
note: `currentMiniBlock` is managing to exceed `64`, which is causing the array pointer to go out of bounds